client/src/components/Personal/Personal.js

```diff
@@ -0,0 +1,128 @@
+import React, { Component } from 'react';
+import { reduxForm, Field } from 'redux-form';
+import validate from '../utils/ValidateWorkflow';
+import FormFields from './FormFields';
+import _ from 'lodash';
+import { fetchUser } from '../../actions/index';
+import { connect } from 'react-redux';
+import * as actions from '../../actions/index';
+import { withRouter } from 'react-router-dom';
+import { ToastContainer, toast } from 'react-toastify';
+import { Link } from 'react-router-dom';
+
+class Personal extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            originalForm: null, // Null at initial state. We wait for redux-form to give us the second value
+            firstRender: true
+        }
+    }
+
+    componentWillReceiveProps({ formValues }) {
+        var { firstRender } = this.state;
+
+        // The redux-form reducer will retrieve our personalForm on the second render
+        if (typeof formValues.values !== "undefined") {
+            formValues.values.formType = "profile";
```

Comment:
Redux-form will always give us an "undefined" as the initial reducer value so we have to handle this.

A user will go to the Profile page, have pre-filled information (because we retrieve the user model via authReducer) and that pre-filled information will fill out the personalForm reducer (redux-form). We want to retrieve this personalForm reducer to add a new field to it: formType of value "profile".

This will tell the action creator what situation to handle, whether the form is coming from the profile page or the workflow.